bots/flakes-refresh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import sys
+import time
+import os
+import urllib
+import socket
+import json
+
+sys.dont_write_bytecode = True
+
+import task
+
+NUMBER_OPEN_ISSUES = 7            # How many issues do we want to have open at a given time?
+
+# How far back does our data go?  If a flake gets fixed but is still
+# flaky after this many days, the bots open another issue.
+
+WINDOW_DAYS = 90
+
+# This parses the output JSONL format discussed here, where various
+# values are grouped:
+#
+# https://github.com/cockpit-project/cockpituous/blob/master/learn/README.md
```

Comment:
This doesn't describe the data format of statistics. One entry looks like this:
```json
{"test": [["testMotd (check_system_info.TestSystemInfo)", 849]], "status": [["success", 849]], "tracker": [[null, 849]], "context": [["verify/debian-stable", 849]], "merged": [[true, 467], [false, 311], [null, 71]], "date": ["2018-07-16T08:25:34Z", "2018-11-09T16:26:37Z"]}
```

What do these mean? I suppose the number is the count of test runs, but e. g. what does `tracker` and `merged[2]`  mean which is `[null, 71]`?

Reply:
Now that we've the most recent changes to cockpituous learn directory, this documentation is now updated. Does it help?

Reply:
It does, thanks. So `merged == null` means "missing or unknown", so probably from PRs which either got merged manually (although we stopped doing that now) or not at all.